src/Interpreters/ExecuteScalarSubqueriesVisitor.cpp

```diff
@@ -108,10 +110,21 @@ void ExecuteScalarSubqueriesMatcher::visit(const ASTSubquery & subquery, ASTPtr
             {
                 if (column.column->empty())
                 {
+                    if (const auto * const_column = typeid_cast<const ColumnConst *>(column.column.get()))
+                    {
+                        /// Even in case of only_analyze there can be real values, if subquery is simple, i.e.
+                        ///     (SELECT 1) as subquery
+                        /// That value is stored in the data column, and it will be used instead of default.
+                        suitable_for_const_folding = !const_column->getDataColumn().empty();
```

Comment:
Can it ever be empty? The comment for `ColumnConst::getValue()` seems to suggest that no:

```
    /// The constant value. It is valid even if the size of the column is 0.
    template <typename T>
    T getValue() const { return getField().safeGet<NearestFieldType<T>>(); }
```

Reply:
>Can it ever be empty? The comment for ColumnConst::getValue() seems to suggest that no:

Indeed, it seems that enough to check for ColumnConst. Thanks, nice catch